src/reconcile/rec_visibility.c

```diff
@@ -252,6 +257,81 @@ __timestamp_out_of_order_fix(WT_SESSION_IMPL *session, WT_TIME_WINDOW *select_tw
     return (false);
 }
 
+/*
+ * __rec_validate_upd_chain --
+ *     Check the update chain for conditions that would prevent it's insertion into the history
+ *     store. Return EBUSY if the update chain cannot be inserted into the history store at this
+ *     time.
+ */
+static int
+__rec_validate_upd_chain(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *select_upd,
+  WT_TIME_WINDOW *select_tw, WT_CELL_UNPACK_KV *vpack)
+{
+    WT_DECL_RET;
+    WT_UPDATE *upd;
+    wt_timestamp_t current_ts;
+    bool seen_ondisk_value;
+
+    upd = NULL;
+    current_ts = WT_TS_NONE;
+    seen_ondisk_value = false;
+
+    /*
+     * If we're not in eviction any history store insertion is fine, if we are in eviction and a
+     * checkpoint isn't running then history store insertions are also fine.
+     */
+    if (!F_ISSET(r, WT_REC_EVICT) || !F_ISSET(r, WT_REC_CHECKPOINT_RUNNING))
+        return (0);
+
+    /*
+     * The selected time window may contain information that isn't visible given the selected
+     * updated, as such we have to check it separately. This is true when there is a tombstone ahead
+     * of the selected update.
+     */
+    if (select_tw->stop_ts < select_tw->start_ts)
+        WT_ERR(EBUSY);
+
+    /*
+     * There is no selected update to go to disk as such we don't need to check the updates
+     * following it.
+     */
+    if (select_upd == NULL)
+        return (0);
+
+    /* Set our variables correctly now that we are sure there's an update. */
+    upd = select_upd->next;
+    current_ts = select_upd->start_ts;
+
+    /* Loop forward from update after the selected on-page update. */
+    for (; upd != NULL; upd = upd->next) {
+        /* Validate that the updates older than us have older timestamps. */
+        if (current_ts < upd->start_ts)
+            WT_ERR(EBUSY);
+
+        /*
+         * If we see transactions id's that belong to the on disk value we don't need to check it
+         * later.
+         */
+        if (vpack != NULL &&
+          (upd->txnid == vpack->tw.stop_txn || upd->txnid == vpack->tw.start_txn))
```

Comment:
I don't think this is a reliable check as we may have cleared the transaction ids on the disk. Shall we just always check the ondisk value if it's not null?

Reply:
I don't remember when transaction IDs are cleared if they are from the previous session due to write generations. Does this code take that into account? i.e: if we are comparing transaction IDs that are read from on-disk cells, is this safely ignoring obsolete transaction IDs?

Reply:
I'm also concerned about this code. I'll leave @luke-pearson to answer this question.

Reply:
Well given that transaction id's can be cleared it's an unreliable check but isn't it possible that the on page update is in the middle of the update chain and if we check it incorrectly against an update older than it we could mistake that as an out of order timestamp. This could be the case in checkpoint reconciliation.

Reply:
We've agreed to remove this check as the WT_UPDATE_RESTORED_FROM_DS flag will tell us if the update has been restored and if it hasn't we can check against the on disk value I'll add a comment explaining why that's okay.